content/en/user-guide/tools/transparent-endpoint-injection/dns-server.md

```diff
@@ -82,60 +120,114 @@ Disabling SSL validation may have undesired side effects and security implicatio
 Make sure to use this only for local testing, and never in production.
 {{< /alert >}}
 
+
+## Customizing internal endpoint resolution
+
+The DNS name `localhost.localstack.cloud` (and any subdomains like `mybucket.s3.localhost.localstack.cloud`) is used internally in LocalStack to route requests, e.g., between a Lambda container and the LocalStack APIs.
+
+Please refer to the steps in the [Route53 docs]({{< ref "route53" >}}) for more details on how the internal DNS name can be customized.
+
+
+## DNS rebind protection
+
+If you rely on your local network's DNS, your router/DNS server might block requests due to the DNS Rebind Protection.
+This feature is enabled by default in pfSense, OPNSense, OpenWRT, AVM FritzBox, and potentially also other devices.
+Some of the vendors might allow upstream responses in the 127.0.0.0/8 range (like OpenWRT).
+
+{{< alert title="Note" >}}
+If you are using the LocalStack DNS server, DNS rebind protection should not cause any issues.
+{{< / alert >}}
+
+You can check if your DNS setup works correctly by resolving a subdomain of `localhost.localstack.cloud`:
+{{< command "hl_lines=16">}}
+$ dig test.localhost.localstack.cloud
+
+; <<>> DiG 9.16.8-Ubuntu <<>> test.localhost.localstack.cloud
+;; global options: +cmd
+;; Got answer:
+;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 45150
+;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1
+
+;; OPT PSEUDOSECTION:
+; EDNS: version: 0, flags:; udp: 65494
+;; QUESTION SECTION:
+;test.localhost.localstack.cloud. IN	A
+
+;; ANSWER SECTION:
+test.localhost.localstack.cloud. 10786 IN CNAME	localhost.localstack.cloud.
+localhost.localstack.cloud. 389	IN	A	127.0.0.1
+
+;; Query time: 16 msec
+;; SERVER: 127.0.0.53#53(127.0.0.53)
+;; WHEN: Fr Jän 14 11:23:12 CET 2022
+;; MSG SIZE  rcvd: 90
+{{< /command >}}
+
+If the DNS resolves the subdomain to your localhost (127.0.0.1), your setup is working.
+If not, please check the configuration of your router / DNS if the Rebind Protection is active or [enable the LocalStack DNS on your system]({{< ref "dns-server#system-dns-configuration" >}}).
+
+
 ## System DNS configuration
 
-In order to use transparent execution mode, the system needs to be configured to use the predefined DNS server.
-This is necessary if you want to test code running directly on your system against LocalStack, instead of AWS.
-The configuration depends on the operating system.
+If you wish to use the DNS server on your host system, you need to expose the LocalStack DNS server and configure your operating system.
+This is necessary if you want to test unmodified application code directly on your system against LocalStack and cannot configure the endpoint URL.
 
 {{< alert title="Warning" color="warning">}}
 Please be careful when changing the network configuration on your system, as this may have undesired side effects.
+Restore the default configuration after testing.
```

Comment:
```suggestion
Remember to restore the default configuration after testing.
```

perhaps?

Reply:
better 👍 
or even more immediately actionable: "Remember to save the default configuration and restore it after testing.